app/models/concerns/has_user_search.rb

```diff
@@ -84,7 +84,11 @@ def self.search(params, timezone, current_full_address)
                         end
 
           end_query = if params[:end_time]
```

Comment:
You can combine the "0:00" and nil checks since they have the same result.

Reply:
nil check use different time zone "UTC". Current behavior will be broken if i change it to end_of_day instead of the timezone chosen by the user.